### DISCUSSION TITLE
FIPs repo directory for supporting information

### DISCUSSION THREAD

**Discussion by anorth** [filecoin-project/FIPs#275] [Archive]
I propose that we add a place in the FIPs repo for documents supporting the improvement proposals themselves.

Each FIP document has a relatively strict structure and benefits from being concise. Often, the decisions expressed in a FIP are the result of a significant amount of data, analysis, modelling etc that would overwhelm the document if included inline. At present, we sometimes link to external resources ([one example](https://github.com/filecoin-project/FIPs/pull/273/files)), but this has some problems:
- We can't generally guarantee the persistence of those resources or the links to them
- The resources may have varying access control mechanisms (e.g. HackMD, Google Docs, other repos) and be unavailable to the public (even unintentionally)
- May have no history/provenance tracking, may change after the fact, etc

Resources checked into the FIPs repo will have the same persistence and access control mechanisms as the proposals themselves. They're easily discoverable even if not linked. The source of the information is clear and public.

I think we should be pretty free with the formats of resources that could be checked in. I would expect to see text or PDF documents, data sets, spreadsheets, code in various languages/environments, etc.

As a concrete proposal, I suggest a directory tree `/resources/FIP-xxx/...`. FIP documents may then link to resources with relative URLs, which should work both in the GitHub UI and other renderings of its content.

cc @Kubuxu

**Comment by Kubuxu** [comment]
I think it is a good idea.

One note is that we should probably set a reasonable size limit for supporting resources checked into the repo (before someone tries checking in 20MB of node_modules or some binaries).

An alternative option would be requiring all supporting resources to be IPFS based and having them fetched and stored using CI.
The PDF you linked is in process of being stored on Filecoin, I will share DealID when it lands.

**Comment by kaitlin-beegle** [comment]
I also agree; I think this is a great idea. 

One small nit: much like how we have a "FIPs" folder, with files for each FIP housed within, I think we should have a single "Resources" folder, with subfolders for any/each FIP that has supporting materials. 

It just makes the repo cleaner.  I also reviewed the initial PR and submitted this feedback as part of my review.

**Reply by Kubuxu** [reply]
I think Anorth's worry is that some FIPs might have way more resources than just one file, and while it might make little sense to create a directory for one file, it clearly links it to FIP-0019 (which isn't that clear without it) and makes the view unified independent whether a FIP has 1 resource file or 20.

**Reply by Kubuxu** [reply]
We might have misunderstood each other. In Alex's PR there is a directory `resources` with a sub-directory `fip-0019` in which there is the file.

**Comment by anorth** [comment]
Implemented in #281. 

~Is there some way to close the discussion? Not lock it, but make it less prominent in the index?~
